### DISCUSSION TITLE
How to access component functions from layout component

### DISCUSSION THREAD

**Discussion by e4rthdog** [quasarframework/quasar#14176] [General - Components / Directives / etc] [answered]
I am looking for a way to access component functions from within the layout vue component.

I can't use component refs because the page is rendered with router-view.

Any ideas?

**Comment by metalsadman** [comment]
you can use template refs on router-view

**Reply by e4rthdog** [reply]
Can you elaborate a little bit? how would the router-view know which page i am rendering? Can you point me to a documentation about that?

Thanks

**Comment by e4rthdog** [comment, marked as answer]
I found it. the solution was to utilize the slots in router-view like this:

```
      <router-view v-slot="{ Component }">
        <component ref="view" :is="Component" />
      </router-view>
```

And then i can reference it like this: `$refs.view.doSomething()`
